qcodes/instrument/parameter.py

```diff
@@ -1798,3 +1801,33 @@ def set_raw(self, value: Union[int, float]) -> None:
 
         self._save_val(value)
         self._wrapped_parameter.set(instrument_value)
+
+
+class DelegateParameter(Parameter):
```

Comment:
Some time ago i implemented a version of this guy [here](https://github.com/kouwenhovenlab/v0_characterization/blob/461a622dd0140c4a00eb7ccfb74584086839bedb/v0_utils/qcodes_tools.py#L146) with support of snapshotting. Could you check it out and, perhaps, use some of the stuff here?

Reply:
sure

Reply:
Oh, great I was looking for it but could not find it.